### PR TITLE
feat: #2745 add support of repeat transfer

### PR DESCRIPTION
**Pull request by gamingumar**
### What does this PR?

Add repeat transfer feature

### Where should the reviewer start?

Repeat Transfer button added in Wallet Screen

### Issue number

https://github.com/ecency/ecency-mobile/issues/2745

### Screenshots/Video

https://github.com/ecency/ecency-mobile/assets/5675313/25be86dd-8b3d-435c-85d0-7ffdc6d67a25

### CODE REVIEW

**Review by noumantahir — CHANGES_REQUESTED** [review]
Kudos! overall great job making your way around the code, left some comments in code, functionality wise we are good.

Just need some code cleaning and we should be good...

**Comment by noumantahir** [comment]
- [ ] One test we can try is to check how app would response in case of `initialAmount` from repeat activity exceeding available asset `balance`

File: src/screens/assetDetails/children/activitiesList.tsx [hunk 1]
```diff
@@ -31,35 +32,48 @@ const ActivitiesList = ({
...
     }
-   
-    
-  }
+  };
 
-  const _renderActivityItem = ({ item, index }) => {
+  const _onRepeatPress = async (item: CoinActivity) => {
+    if (onActionPress) {
+      onActionPress('transfer_token', item);
```
[the remark this answers is not in the document]
Comment (gamingumar) [reply src/screens/assetDetails/children/activitiesList.tsx:61]: UPDATED

File: src/screens/transfer/screen/transferScreen.js [hunk 2]
```diff
@@ -62,10 +63,18 @@ const TransferView = ({
       ? 'esteem.app'
       : transferType === TransferTypes.DELEGATE_SPK
       ? SPK_NODE_ECENCY
+      : transferType === TransferTypes.TRANSFER_TOKEN && transactionData
+      ? transactionData.toUsername
+      : '',
+  );
+  const [amount, setAmount] = useState(transactionData ? transactionData.value?.split(' ')[0] : '');
```
[the remark this answers is not in the document]
Comment (gamingumar) [reply src/screens/transfer/screen/transferScreen.js:70]: updated

File: src/components/basicUIElements/view/walletLineItem/walletLineItemView.tsx [hunk 3]
```diff
@@ -102,18 +101,35 @@ const WalletLineItem = ({
...
+            isLoading={cancelling}
+          />
         )}
 
+        {!!onRepeatPress && (
+          <IconButton
+            backgroundColor="transparent"
+            name="repeat"
+            iconType="FontAwesome"
+            size={20}
```
[the remark this answers is not in the document]
Comment (gamingumar) [reply src/components/basicUIElements/view/walletLineItem/walletLineItemView.tsx:124]: updated

File: src/screens/assetDetails/screen/assetDetailsScreen.tsx [hunk 4]
```diff
@@ -148,6 +148,16 @@ const AssetDetailsScreen = ({ navigation, route }: AssetDetailsScreenProps) => {
       };
     }
 
+    if (extraParams) {
+      navigateParams = {
+        ...navigateParams,
+        itemData: {
+          ...extraParams,
```
[the remark this answers is not in the document]
Comment (gamingumar) [reply src/screens/assetDetails/screen/assetDetailsScreen.tsx]: updated

**Comment by noumantahir** [comment]
Screenshot 2023-09-12 at 11 58 35 AM

**Review by noumantahir — COMMENTED** [review]
lgtm

**Review by noumantahir — APPROVED** [review]

**Review by feruzm — APPROVED** [review]
lgtm

**Pull request merged**